cordapp-upgrades/contracts/v1-contract/src/main/kotlin/net/corda/examples/obligation/contract/Obligation.kt

```diff
@@ -0,0 +1,31 @@
+package net.corda.examples.obligation.contract
+
+import net.corda.core.contracts.Amount
+import net.corda.core.contracts.BelongsToContract
+import net.corda.core.contracts.LinearState
+import net.corda.core.contracts.UniqueIdentifier
+import net.corda.core.crypto.NullKeys
+import net.corda.core.identity.AbstractParty
+import net.corda.core.identity.Party
+import net.corda.core.utilities.toBase58String
+import java.util.*
+
+@BelongsToContract(ObligationContract::class)
+data class Obligation(val amount: Amount<Currency>,
+                      val lender: AbstractParty,
+                      val borrower: AbstractParty,
+                      val paid: Amount<Currency> = Amount(0, amount.token),
```

Comment:
Not relevant to the task at hand, but the c'tor for this class should make sure the currency of `paid` is the same as the currency of `amount`

Reply:
I've updated this both here and in the original app.